aten/src/ATen/native/cuda/SoftMax.cu

```diff
@@ -198,7 +198,11 @@ T spatialBlockReduceX(T *shared, T val) {
 }
 
 template <typename scalar_t, typename accscalar_t, typename outscalar_t, template<typename, typename, typename> class Epilogue>
-__global__ void cunn_SpatialSoftMaxForward(
+__global__ void
+#if __CUDA_ARCH__ == 620
+  __launch_bounds__(512, 1)
```

Comment:
The runtime block size for this kernel [comes from](https://github.com/pytorch/pytorch/blob/ebd727fc9f4c1f273b2395b9942cd2dc1d775e93/aten/src/ATen/native/cuda/SoftMax.cu#L104-L107)
```C++
inline dim3 SpatialSoftMax_getBlockSize(
  uint64_t outer_size, uint64_t dim_size, uint64_t inner_size) {
  uint32_t inner_threads = inner_size;
  inner_threads = std::min(inner_threads, static_cast<uint32_t>(max_threads));
  ...
```